src/main/java/org/jacp/controller/CompetitionController.java

```diff
@@ -77,6 +77,16 @@ public ResponseEntity<CompetitionDto> startCompetition(@PathVariable Long compet
         return ResponseEntity.ok(competitionDto);
     }
 
+    @PostMapping("/{competitionId}/join")
+    public ResponseEntity<CompetitionDto> joinParticipants(@PathVariable Long competitionId,
+                                                           @RequestParam Long participantId) {
+        CompetitionDto competitionDto =
+                mapper.toCompetitionEntityByCompetitionDto(competitionService.getCompetitionEntity(competitionId));
+        competitionDto.joinParticipant(participantId);
+        competitionService.create(mapper.toCompetitionEntity(competitionDto));
```

Comment:
It's not a common practice to transfer entity object to dto and after some changes move it back.
It's much easier to just have a one more service method: `competitionService.joinParticipant(competitionId, participantId) ` and inside the method just do all required changes on the entity.

Reply:
Fixed